# since mosquitto 1.6.11 pthread_setname_np is required, but not supported with glibc < 2.12
# this is for naming threads only as introduced with https://github.com/eclipse/mosquitto/commit/bd27935ff63613301e485c0ce9e7e9b4e67e6d07
# and has no effect to the theading support in mosquitto.

--- lib/thread_mosq.c.org	2020-08-19 13:55:03.000000000 +0000
+++ lib/thread_mosq.c	2020-09-29 07:13:18.196407777 +0000
@@ -41,13 +41,6 @@
 
 	mosq->threaded = mosq_ts_self;
 	if(!pthread_create(&mosq->thread_id, NULL, mosquitto__thread_main, mosq)){
-#if defined(__linux__)
-		pthread_setname_np(mosq->thread_id, "mosquitto loop");
-#elif defined(__NetBSD__)
-		pthread_setname_np(mosq->thread_id, "%s", "mosquitto loop");
-#elif defined(__FreeBSD__) || defined(__OpenBSD__)
-		pthread_set_name_np(mosq->thread_id, "mosquitto loop");
-#endif
 		return MOSQ_ERR_SUCCESS;
 	}else{
 		return MOSQ_ERR_ERRNO;
